app/Models/Attribute.php

```diff
@@ -365,4 +367,43 @@ public function listRangeText(): string
     {
         return implode(', ', $this->listRange);
     }
+
+    /**
+     * Get the value used to index the model.
+     *
+     * @return mixed
+     */
+    public function getScoutKey()
+    {
+        return $this->getTable() . '_' . $this->id;
+    }
+
+    /**
+     * Get the name of the index associated with the model.
+     */
+    public function searchableAs(): string
+    {
+        return 'entities';
+    }
+
+    protected function makeAllSearchableUsing($query)
+    {
+        return $query
+        ->leftJoin('entities', 'attributes.entity_id', '=', 'entities.id');
+    }
+
+    public function toSearchableArray()
+    {
+        $array = $this->toArray();
```

Comment:
Why do we need to do toArray() and not directly `return [ 'campaign_id' => $this->entity->campaign_id, etc]`? This seems like a waste of processing power to make everything an array (also makes ctrl+click to go to the definition not work in phpstorm)